### PR TITLE
Kubernetes : new Deployment API and adapted DeployWatcher (tests)

### PR DESCRIPTION
**Pull request by sbrnunes**
PR created in top of #698. Tests have been fixed and a few new tests have been added to the DeployWatcher.

This PR should be reviewed when #698 has been merged, in order to reduce the scope of the changes.

/cc @msufa, @henders, @jonmoter

### Tasks
 - [x] DeployWatcher should track events for any Release regarding the given Project
 - [x] DeployWatcher should synchronise the DB with the current state in the cluster before it starts watching for events
 - [x] When creating a new Release, if there's a ongoing Deployment, the current watcher must be stopped before a watcher is created (to avoid concurrent updates)
 - [x] Tests

### CODE REVIEW

File: plugins/kubernetes/test/models/watchers/cluster_pod_watcher_test.rb [hunk 1]
```diff
@@ -66,54 +71,65 @@
...
-      assert_equal(:cluster_pod_watcher_1,
-                   Watchers::ClusterPodWatcher.pod_watcher_symbol(kubernetes_clusters(:test_cluster)))
+    let(:cluster) { kubernetes_clusters(:test_cluster) }
+
+    it 'builds a correct ClusterPodWatcher name' do
+      assert_equal(cluster_name(cluster).to_sym, Watchers::ClusterPodWatcher.watcher_symbol(cluster))
+    end
+
+    it 'builds a correct ClusterPodErrorWatcher watcher name' do
+      assert_equal(cluster_name(cluster).to_sym, Watchers::ClusterPodWatcher.watcher_symbol(cluster))
```
Comment (msufa) [inline plugins/kubernetes/test/models/watchers/cluster_pod_watcher_test.rb]: Copy/Paste? :smiley_cat: I think you wanted to test `ClusterPodErrorWatcher`

Comment (msufa) [inline plugins/kubernetes/test/models/watchers/cluster_pod_watcher_test.rb]: Also, I don't think this test file is a good place for that test.

Comment (sbrnunes) [inline plugins/kubernetes/test/models/watchers/cluster_pod_watcher_test.rb]: You caught me. Damn.

Nah, you're right, I was not thinking when I did this. As you said... copy/paste... Shame on me. :)

**Comment by msufa** [comment]
:+1:

**Pull request merged**